internal/services/apimanagement/api_management_workspace_policy_resource.go

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"fmt"
+	"html"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2024-05-01/workspace"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2024-05-01/workspacepolicy"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ApiManagementWorkspacePolicyModel struct {
+	ApiManagementWorkspaceId string `tfschema:"api_management_workspace_id"`
+	XmlContent               string `tfschema:"xml_content"`
+	XmlLink                  string `tfschema:"xml_link"`
+}
+
+type ApiManagementWorkspacePolicyResource struct{}
+
+var _ sdk.ResourceWithUpdate = ApiManagementWorkspacePolicyResource{}
+
+func (r ApiManagementWorkspacePolicyResource) ResourceType() string {
+	return "azurerm_api_management_workspace_policy"
+}
+
+func (r ApiManagementWorkspacePolicyResource) ModelObject() interface{} {
+	return &ApiManagementWorkspacePolicyModel{}
+}
+
+func (r ApiManagementWorkspacePolicyResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return workspacepolicy.ValidateWorkspaceID
+}
+
+func (r ApiManagementWorkspacePolicyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"api_management_workspace_id": commonschema.ResourceIDReferenceRequiredForceNew(&workspace.WorkspaceId{}),
+
+		"xml_content": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			// NOTE: O+C when `xml_link` is provided, the API downloads it into `xml_content`.
+			// O+C ensures Terraform state is populated correctly and prevents spurious diffs.
+			Computed:         true,
+			ExactlyOneOf:     []string{"xml_link", "xml_content"},
+			DiffSuppressFunc: XmlWithDotNetInterpolationsDiffSuppress,
+		},
+
+		"xml_link": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ExactlyOneOf: []string{"xml_link", "xml_content"},
+			ValidateFunc: validation.StringIsNotEmpty,
+			DiffSuppressFunc: func(k, old, new string, d *pluginsdk.ResourceData) bool {
```

Comment:
could you set `xml_link` with the configuration value in `read` function instead of using `DiffSuppressFunc`?